### DISCUSSION TITLE
Manage Metadata with TiDB rather than MySQL

### DISCUSSION THREAD

**Discussion by t0t07** [milvus-io/milvus#4842] [Q&A and General discussion] [answered]
According to https://milvus.io/docs/v1.0.0/data_manage.md, Milvus supports SQLite and MySQL for metadata management.

I wonder if I can use TiDB for the same purpose, given it is [compatible](https://docs.pingcap.com/tidb/stable/mysql-compatibility) with MySQL?

Thanks.

**Comment by yhmo** [comment, marked as answer]
We will continue maintain the v1.0.x and fix critical issues. But for the metadata backend(Postgresql, TiDB, etc.), currently we have no enough resource to implement this kind of feature.

**Reply by t0t07** [reply]
@yhmo Thanks for the reply. Just to double check, is managing metadata with TiDB:
- Not yet tested, or
- Proved to be not working?

I ask because as said in the question, TiDB is compatible with MySQL to some extent, therefore in some scenarios people can simply treat the TiDB as a MySQL.

Thank you!

**Reply by yhmo** [reply]
In Milvus source code, metadata backend interface is defined in this header: /milvus/core/src/db/meta/Meta.h
There are about 30 methods in the interface. Now there are two kinds implements: mysql and sqlite. To support a new kind metadata backend, developers need to declare a class inherit from the interface and implement the methods.

Milvus use a thirdparty "mysql++" to connect to mysql server.

I just check the PingCAP website: https://docs.pingcap.com/zh/tidb/v5.0/connectors-and-apis. From this page we know: all of MySQL connector is supported by TiDB.
So, I think milvus can work with TiDB. But we didn't test it.

**Reply by t0t07** [reply]
@yhmo Thanks for the clear explanation. I will try running Milvus with TiDB and update what I find.

**Reply by t0t07** [reply]
Summary: According to what TiDB claim, we should be able to manage metadata with TiDB rather than MySQL. However, this is not tested yet.
